autoreject/utils.py

```diff
@@ -248,17 +248,29 @@ def get_picks_bad_good(info):
     if len(picks_meg) == 0 or len(picks_bad) == 0:
         return
 
+    # we need to make sure that only meg channels are passed here
+    # as the MNE interpolation code is not fogriving.
+    # This is why we picked the info.
     mapping = _fast_map_meg_channels(
-        picked_info, pick_from=picks_good, pick_to=picks_bad,
+        picked_info.copy(), pick_from=picks_good, pick_to=picks_bad,
         mode=mode)
-
-    # recompute picks_good, picks_bad
-    _, _, picks_bad_ = get_picks_bad_good(inst.info)
-    # mapping_ = np.empty(len(ch_names), len(ch_))
+    # the downside is that the mapping matrix now does not match
+    # the unpicked info of the data.
+    # Since we may have picked the info, we need to double map
+    # the indices.
+    _, picks_good_, picks_bad_orig = get_picks_bad_good(inst.info.copy())
     ch_names_a = [picked_info['ch_names'][pp] for pp in picks_bad]
-    ch_names_b = [inst.info['ch_names'][pp] for pp in picks_bad_]
+    ch_names_b = [inst.info['ch_names'][pp] for pp in picks_bad_orig]
     assert ch_names_a == ch_names_b
-    _do_interp_dots(inst, mapping, picks_good, picks_bad_)
+    if not inst_picked:
+        picks_good_ = [pp for pp in picks if pp in picks_good_]
+    assert len(picks_good_) == len(picks_good)
+
+    # XXX all trouble is probably here
+    inst._data[:, picks_bad_orig, :] = (
+        # multiply correctly with interpolation matrix
+        np.einsum('ij,xjy->xiy',  # subpick data
+                  mapping, inst._data[:, picks_good_]))
```

Comment:
@jasmainak can you take a look? I think somehow the remapping between the picked info from where we get the mapping the the unpicked instance is not correct. Tried to leave some comments. Debugging this is difficult as long as we don't  have a regression test / test against ground truth. Currently the symptom is imply that the right bad channel is not repaired.

See here the output from plot_auto_repair.py

<img width="567" alt="screenshot 2017-06-25 22 51 37" src="https://user-images.githubusercontent.com/1908618/27520367-6fdc758c-5a0a-11e7-9669-9c817832313f.png">

However, the code before seems to work as we have clearly a bad channel here:

<img width="952" alt="screenshot 2017-06-25 22 56 15" src="https://user-images.githubusercontent.com/1908618/27520373-88f89ab4-5a0a-11e7-97b9-3b47374522fe.png">